src/main/java/step3/RacingCarGame.java

```diff
@@ -0,0 +1,51 @@
+package step3;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class RacingCarGame {
+    private List<RacingCar> racingCarList;
+    private Attempt attempt;
+
+    public boolean ready(int carCount, int attemptCount) {
+        racingCarList = new ArrayList<>(carCount);
+        while(racingCarList.size() != carCount) {
+            racingCarList.add(new RacingCar());
+        }
+
+        attempt = new Attempt(attemptCount);
+        if(isRacingCarReady(carCount) && isAttemptReady(attemptCount)) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isRacingCarReady(int carCount) {
+        if(racingCarList.size() == carCount) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isAttemptReady(int attemptCount) {
+        return attempt.isEqualNumber(attemptCount);
+    }
+
+    public List<RacingCar> getRacingCarList() {
+        return racingCarList;
+    }
+
+    public Attempt getAttempt() {
+        return attempt;
+    }
+
+    public void play() {
+        Iterator iterator = racingCarList.iterator();
+        while(iterator.hasNext()) {
+            RacingCar racingCar = (RacingCar) iterator.next();
+            racingCar.move();
+        }
+        attempt.make();
```

Comment:
```suggestion
        for (RacingCar racingCar : racingCarList) {
            racingCar.move();
        }
        attempt.make();
```
요렇게도 쓸 수 있습니다.:)
또는`Iterator<RacingCar> iterator = racingCarList.iterator();` 로 타입을 지정해주면 next()에서 타입캐스팅을 안해줘도 될 것 같아요